trie/triedb/pathdb/history.go

```diff
@@ -576,8 +576,10 @@ func truncateFromHead(db ethdb.Batcher, freezer *rawdb.ResettableFreezer, nhead
 		return 0, err
 	}
 	// Ensure that the truncation target falls within the specified range.
-	if ohead < nhead || nhead < otail {
-		return 0, fmt.Errorf("out of range, tail: %d, head: %d, target: %d", otail, ohead, nhead)
+	if nhead != 0 { // truncating to zero is always possible
```

Comment:
I think this is not a sufficient fix. The code further down: 

```golang
	// Load the meta objects in range [nhead+1, ohead]
	blobs, err := rawdb.ReadStateHistoryMetaList(freezer, nhead+1, ohead-nhead)
	if err != nil {
		return 0, err
	}
	batch := db.NewBatch()
	for _, blob := range blobs {
		var m meta
		if err := m.decode(blob); err != nil {
			return 0, err
		}
		rawdb.DeleteStateID(batch, m.root)
	}
```
The call to `ReadStateHistoryMetaList` will read the _entire_ chunk of data into memory, since it internally does 
```
	return db.AncientRange(stateHistoryMeta, start-1, count, 0)
```
using `0` as the maxbytes (` if maxBytes is not specified, 'count' items will be returned if they are present`). 

Also, I'm not sure how this would behave if we try to load 5M, but 0-4M are already gone. 

A better fix might be to something like this: 
```
	if nhead != 0 { // truncating to zero is always possible
		ohead, err = freezer.TruncateHead(0)
		if err != nil {
			return 0, err
		}
		return int(ohead - nhead), nil
	}
```
But if resetting to `0`, we should also set back the `tail` to zero. Hm... @rjl493456442 thoughts?